sandbox/apparmor/apparmor.go

```diff
@@ -316,27 +321,38 @@ func probeKernelFeatures() ([]string, error) {
 }
 
 func probeParserFeatures() ([]string, error) {
-	parser, err := findAppArmorParser()
+	parser, internal, err := FindAppArmorParser()
 	if err != nil {
 		return []string{}, err
 	}
 	features := make([]string, 0, 1)
 	if tryAppArmorParserFeature(parser, "change_profile unsafe /**,") {
 		features = append(features, "unsafe")
 	}
+	if internal {
+		features = append(features, "snapd-internal")
+	}
 	sort.Strings(features)
 	return features, nil
 }
 
-// findAppArmorParser returns the path of the apparmor_parser binary if one is found.
-func findAppArmorParser() (string, error) {
+// FindAppArmorParser returns the path of the apparmor_parser binary if one is found.
+func FindAppArmorParser() (string, bool, error) {
+	// first see if we have our own internal copy
+	if path, err := snapdtool.InternalToolPath("apparmor_parser"); err == nil {
+		if _, err := os.Stat(path); err == nil {
```

Comment:
This could use `osutil.IsExecutable()` which doubles an existing check in InternalToolPath for specific scenarios, but at least we'll know if the permission bits are right in other scenarios.

Reply:
Good point - added in https://github.com/snapcore/snapd/pull/9939/commits/35acd6d3265a3aa7489c1e8e45735038807073bb